superset/db_engine_specs/bigquery.py

```diff
@@ -282,3 +313,40 @@ def df_to_sql(
                 to_gbq_kwargs[key] = to_sql_kwargs[key]
 
         pandas_gbq.to_gbq(df, **to_gbq_kwargs)
+
+    @classmethod
+    def build_sqlalchemy_uri(
+        cls, _: BigQueryParametersType, encrypted_extra: Optional[Dict[str, str]] = None
+    ) -> str:
+        if encrypted_extra:
+            project_id = encrypted_extra.get("project_id")
+            return f"{cls.drivername}://{project_id}"
+
+        raise SupersetGenericDBErrorException(
+            message="Big Query encrypted_extra is not available",
+        )
+
+    @classmethod
+    def get_parameters_from_uri(cls, _: str) -> Any:
+        # BigQuery doesn't have parameters
+        return None
```

Comment:
Ideally:

```
get_parameters_from_uri(build_sqlalchemy_uri(parameters)) == parameters
```

Ie, this should be the opposite of `build_sqlalchemy_uri`.

We need to return `credentials_info` here so that the edit form can show it to the user. This means that we need to modify `get_parameters_from_uri` to also receive `encrypted_extra`.